core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -5050,6 +5050,13 @@ private void checkLiteral2(String expression, String expected) {
   }
 
   @Test void testSelectNull() {
+    String query = "SELECT NULL AS DUMMY FROM \"product\"";
+    final String expected = "SELECT CAST(NULL AS NULL) AS \"DUMMY\"\n"
+            + "FROM \"foodmart\".\"product\"";
+    sql(query).ok(expected);
```

Comment:
Can we promote the `CAST(NULL AS NULL)` to just `NULL` ?

Reply:
Thanks: I missed this improvement using 1.23 as a reference. fixed.